server/lomas_server/app.py

```diff
@@ -47,91 +47,115 @@
+
+    Server initialization is performed (config loading, etc.) and
+    the server state is updated accordingly. This can have potential
+    side effects on the return values of the "depends"
+    functions, which check the server state.
     """
-    # Startup event
+    # Startup
     LOG.info("Startup message")
-    global CONFIG
-    global QUERY_HANDLER
-    global ADMIN_DATABASE
 
-    SERVER_STATE["state"].append("Startup event")
+    # Set some app state
+    app.state.admin_database = None
+    app.state.query_handler = None
 
-    # Load config here
-    LOG.info("Loading config")
-    SERVER_STATE["message"].append("Loading config")
-    CONFIG = get_config()
+    # General server state, can add fields if need be.
+    app.state.server_state = {
+        "state": [],
+        "message": [],
+        "LIVE": False,
+    }
+    app.state.server_state["state"].append("Startup event")
+
+    status_ok = True
+    # Load config
+    try:
+        LOG.info("Loading config")
+        app.state.server_state["message"].append("Loading config")
+        config = get_config()
+    except InternalServerException:
+        LOG.info("Config could not loaded")
+        app.state.server_state["state"].append(CONFIG_NOT_LOADED)
+        app.state.server_state["message"].append(
+            "Server could not be started!"
+        )
+        app.state.server_state["LIVE"] = False
+        status_ok = False
 
     # Fill up user database if in develop mode ONLY
-    if CONFIG.develop_mode:
+    if status_ok and config.develop_mode:
         LOG.info("!! Develop mode ON !!")
-        LOG.info("Creating example user collection")
-
-        args = SimpleNamespace(**vars(CONFIG.admin_database))
-
-        LOG.info("Creating user collection")
-        args.clean = True
-        args.overwrite = True
-        args.path = "/data/collections/user_collection.yaml"
-        create_users_collection(args)
-
-        LOG.info("Creating datasets and metadata collection")
-        args.path = "/data/collections/dataset_collection.yaml"
-        args.overwrite_datasets = True
-        args.overwrite_metadata = True
-        add_datasets(args)
-
-        LOG.info("Empty archives")
-        args.collection = "queries_archives"
-        drop_collection(args)
-
-    # Load users, datasets, etc..
-    LOG.info("Loading admin database")
-    SERVER_STATE["message"].append("Loading admin database")
-    try:
-        ADMIN_DATABASE = database_factory(CONFIG.admin_database)
-    except InternalServerException as e:
-        LOG.exception("Failed at startup:" + str(e))
-        SERVER_STATE["state"].append("Loading user database at Startup failed")
-        SERVER_STATE["message"].append(str(e))
+        LOG.info("Adding demo data to AdminDB")
+        app.state.server_state["message"].append("!! Develop mode ON !!")
+        app.state.server_state["message"].append("Adding demo data to AdminDB")
+        add_demo_data_to_admindb(config)
```

Comment:
not sure for the config argument. shouldn't you remove it since you load it within the function `add_demo_data_to_admindb()` ?

Reply:
ah right